gover/proxy/bypass.go

```diff
@@ -15,17 +15,17 @@ func genPrngSeed(seed uint64) uint64 {
 }
 
 func sniffKey(mSeed, sentMs uint64, packet []byte) uint64 {
-	key := utils.NewPacketKey()
-	buf := make([]byte, len(packet))
+	k := utils.NewMT19937_64()
+	v := be.Uint64(packet)
 	sniff := func(seed uint64) bool {
-		key.GenKey(seed)
-		copy(buf, packet)
-		key.Xor(buf)
-		if be.Uint16(buf) == 0x4567 &&
-			be.Uint16(buf[len(buf)-2:]) == 0x89AB {
-			return true
-		}
-		return false
+		k.Seed(seed)
+		k.Seed(k.Int64())
+		k.Int64()
+		// 0     2     4     6     |       10
+		// 45 67 ?? ?? 00 ?? 00 00 | ?? ?? ?? ??...
+		// ^~~~~ ^~~~~ ^~~~~ ^~~~~~~~~~~~~ ^~~~~~...
+		// MAGIC CMDID H_LEN B_LEN         H_DATA...
+		return v^k.Int64()&0xFFFF0000FF00FFFF == 0x4567000000000000
```

Comment:
It's a very sensible approach, but I had tried to match magic only the beginning of the payload before, and there was a certain chance of failure, so I added a matching magic at the end; is there a way to prove that only detecting the beginning is feasible?

Reply:
FYI, the header length and body length may can be calculated, according to past experience, the header length is always fixed